packages/sapience/src/app/forecast/page.tsx

```diff
@@ -1,301 +1,35 @@
-            {selectedMarket && (
-              <div className="p-4">
-                <PredictForm
-                  marketGroupData={marketGroupData}
-                  marketClassification={marketClassification}
-                  onSuccess={refetchComments}
-                />
-              </div>
-            )}
-          </div>
-          {/* Category Selection Section */}
-          <div className="bg-background">
-            <div
-              className="flex overflow-x-auto"
-              style={{ WebkitOverflowScrolling: 'touch' }}
-              onWheel={(e) => {
-                if (e.deltaY === 0) return;
-                e.currentTarget.scrollLeft += e.deltaY;
-                e.preventDefault();
-              }}
-            >
-              {/* All option - moved to first position */}
-              <button
-                type="button"
-                onClick={() => setSelectedCategory(null)}
-                className={`flex items-center gap-1.5 px-2 py-1.5 transition-colors text-xs whitespace-nowrap border-r border-border border-b-2 ${
-                  selectedCategory === null ? 'border-b-primary' : ''
-                } ${
-                  selectedCategory === null
-                    ? selectedStatusClass
-                    : hoverStatusClass
-                }`}
-              >
-                <div className="rounded-full p-0.5 w-4 h-4 flex items-center justify-center bg-zinc-500/20">
-                  <LayoutGridIcon className="w-2.5 h-2.5 text-zinc-500" />
-                </div>
-                <span className="font-medium">All</span>
-              </button>
+const ForecastPage = async () => {
+  // new query client for the server
+  const serverQC = new QueryClient();
 
-              {/* Selected Question option - only show when a market is selected */}
-              {selectedMarket && (
-                <button
-                  type="button"
-                  onClick={() => setSelectedCategory(SelectableTab.Selected)}
-                  className={`flex items-center gap-1.5 px-2 py-1.5 transition-colors text-xs whitespace-nowrap border-r border-border border-b-2 ${
-                    selectedCategory === SelectableTab.Selected
-                      ? 'border-b-primary'
-                      : ''
-                  } ${
-                    selectedCategory === SelectableTab.Selected
-                      ? selectedStatusClass
-                      : hoverStatusClass
-                  }`}
-                >
-                  <div className="rounded-full p-0.5 w-4 h-4 flex items-center justify-center bg-zinc-500/20">
-                    <FileTextIcon className="w-2.5 h-2.5 text-zinc-500" />
-                  </div>
-                  <span className="font-medium">Selected Question</span>
-                </button>
-              )}
+  // Prefetch enriched market groups data
+  await prefetchEnrichedMarketGroups(serverQC);
+  await prefetchPredictions(serverQC, CONVERGE_SCHEMA_UID);
```

Comment:
should this be in getServerSideProps()?

https://tanstack.com/query/latest/docs/framework/react/guides/ssr
https://nextjs.org/docs/pages/api-reference/functions/get-server-side-props

Reply:
You are viewing the router documentation pages. This projects use App Router.